### PR TITLE
fix: support multiple node urls

### PR DESCRIPTION
**Pull request by kristifidani**
# Pull Request Template

## Motivation and Context

Previously we could specify multiple node urls but now the NetworkType supports only single node_url. We need to update this to support again multiple node urls.

## Summary

Provide a short summary of changes in this pull request.

- Update api types and sdk.
- Update bindings and tests.

Gitlab changes: https://gitlab.inovel.de/zd0006/cryptpay.rs/-/merge_requests/457

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: sdk/bindings/swift/src/lib.rs [hunk 1]
```diff
@@ -203,7 +203,7 @@ pub mod ffi {
         pub amount: f64,
         pub network: String,
         pub status: String,
-        pub explorer_url: String,
+        pub explorer_url: Vec<String>,
```
Comment (antbern) [inline sdk/bindings/swift/src/lib.rs:206]: I do not think we need to support multiple explorer urls, it is only the node urls that we need to support 💯 
```suggestion
        pub explorer_url: String,
```

File: api_types/src/api/networks.rs [hunk 2]
```diff
@@ -3,8 +3,8 @@ use serde::{Deserialize, Serialize};
 #[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
 #[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
 pub enum ApiNetworkType {
-    Evm { node_url: String, chain_id: u64 },
-    Stardust { node_url: String },
+    Evm { node_url: Vec<String>, chain_id: u64 },
```
Comment (antbern) [inline api_types/src/api/networks.rs:6]: Should we also rename the variable to indicate that there are multiple URLs?
```suggestion
    Evm { node_urls: Vec<String>, chain_id: u64 },
```

File: sdk/src/wallet/wallet_user_eth.rs [hunk 3]
```diff
@@ -96,17 +96,17 @@ impl WalletImplEth {
         mnemonic: Mnemonic,
         path: &Path,
         http_provider: RootProvider<Http<Client>>,
-        node_url: String,
+        node_url: Vec<String>,
```
Comment (antbern) [inline sdk/src/wallet/wallet_user_eth.rs:99]: Funny that we now take this as a vector but then later on only use the first element to create the Http Client for Alloy... 😄 I hope alloy has support for multiple node-urls, otherwise this change is kinda meaningless TBH 🌵 @lszetg do you know if alloy supports mutlliple urls?

**Review by antbern — COMMENTED** [review]
Nice, added some comments 💯

File: sdk/src/wallet/wallet_user_eth.rs (hunk 3 above)
Comment (lszetg) [reply sdk/src/wallet/wallet_user_eth.rs:99]: almost there :D 
- https://github.com/alloy-rs/alloy/pull/2135

Comment (antbern) [reply sdk/src/wallet/wallet_user_eth.rs:99]: > almost there :D
> 
> * [feat: `FallbackLayer` transport alloy-rs/alloy#2135](https://github.com/alloy-rs/alloy/pull/2135)

Alright good, otherwise this is not really a useful feature 😄

File: sdk/bindings/swift/src/lib.rs (hunk 1 above)
Comment (kristifidani) [reply sdk/bindings/swift/src/lib.rs]: I saw that we were assigning `node urls` to the `explorer_url` and that's the chain of events that resulted in this change. I reverted this back and added some comments to fix this and use the `Network` type instead of some hardcoded values 😄

File: api_types/src/api/networks.rs (hunk 2 above)
Comment (kristifidani) [reply api_types/src/api/networks.rs]: Done 👍

**Review by antbern — APPROVED** [review]
Looking good! :rocket:

**Review by hauju — APPROVED** [review]

**Comment by hauju** [comment]
LGMT!
